src/main/scala/fundamentals/level04/LogParser.scala

```diff
@@ -104,7 +104,7 @@ object LogParser {
     * scala> showErrorsOverSeverity(logFile, 2)
     * = List("Error 5 (158) some strange error")
     *
-    * Hint: Use `parseLogFile` and `showLogMessage`
+    * Hint: Use `parseLogFile`, `getErrors` and `showLogMessage`
```

Comment:
I'm not sure what advantage using `getErrors` would give you here. `getErrors` still returns a `List[LogMessage]`, which means you'd still have to `filter` or `collect` on each `LogMessage` to pull out the `Error(severity)`. Is there something I'm missing?

Reply:
Where do you check the severity?

Reply:
Ah I see!

What if `getErrors` becomes `getErrorsOverSeverity`?

Or should we just keep `getUnknowns` because this doesn't seem like an improvement anymore?

Reply:
Should we have both maybe? `getUnknowns` and `getErrorsOverSeverity`. Then `showErrorsOverSeverity` just maps `showLogMessage` over `getErrorsOverSeverity`, which promotes good functional composition.

Reply:
I personally think the trickiest part of this `LogParser` exercise is the `parseLog` function. I obviously wasn't there for the last session, but I can imagine people getting stuck on that more than these functions.

Reply:
I’m reluctant to add a new exercise. I think it’s either getUnknowns or getErrorsOverSeverity. 

parseLog is hard! Stili handheld everyone through it by wrapping it all in a try catch. I think for an end of day exercise that’s good enough :)

Reply:
That's fair enough. One option for `parseLog` might be to have a `parseInfo(String, String): KnownLog`, a `parseWarning(String, String): KnownLog` and a `parseError(String, String, String): KnownLog`. Then use those in `parseLog`.

On the note of `getUnknowns` vs `getErrorsOverSeverity`, I'm happy to go with `getErrorsOverSeverity`.

Reply:
I'll push a commit when I get some time to update `getErrors` to `getErrorsOverSeverity`